### DISCUSSION TITLE
PointSetBC() doubt

### DISCUSSION THREAD

**Discussion by guimilhoratti** [lululxvi/deepxde#1432] [Q&A]
Hi all,

I am using boundary_conditions.PointSetBC() in my code, however, I am unsure if I understand how it works. 

It seems that it generates Dirichlet boundary conditions from the exact solution. 

1-Does it use border data from the solution? 
2-Does it calculate boundary loss using each point and its respective solution result? 
3-How can I learn more about it? 

Best Regards,

**Comment by ryabs07** [comment]
Hi,
I think [this example of inverse problem](https://deepxde.readthedocs.io/en/latest/demos/pinn_inverse/diffusion.1d.inverse.html)  is a pretty good for understanding execution of PointSetBC. 

A snippet of code from the example:
```
observe_x = np.vstack((np.linspace(-1, 1, num=10), np.full((10), 1))).T
observe_y = dde.icbc.PointSetBC(observe_x, func(observe_x), component=0)

data = dde.data.TimePDE(
    geomtime,
    pde,
    [bc, ic, observe_y],
    num_domain=40,
    num_boundary=20,
    num_initial=10,
    anchors=observe_x,
    solution=func,
    num_test=10000,
)
```
_**observe_x**_ is an array of input points for which you have observed values **_observe_y_** (you can input just the boundary points if you strictly want to use it for BC).

**Reply by guimilhoratti** [reply]
Hi @ryabs07 

Thank you for your response. 

I think it is clear now. It is the supervised ("extra observations ") part of the the training, right?

Kind regards

**Reply by ryabs07** [reply]
yes exactly. how are the results coming along for you using PointSetBC? For me, the loss for this supervised training portion stops decreasing after couple of hundred iterations and the results are quite bad.

**Reply by guimilhoratti** [reply]
My results are bad and the supervisory training has plateaued after a thousand iterations. Have you tried using a smaller dataset?

**Reply by ryabs07** [reply]
I was trying to use PointSetBC for initial condition only and I had barely 300 points. I face the same issue. I have tried multiple hyperparameters, but nothing seems to work. I have posted the problem in the discussions, hopefully someone experienced can give us some suggestions.

**Reply by guimilhoratti** [reply]
I hope so, local minima make things complicated

**Reply by ryabs07** [reply]
Hi again,
Have you had any success with the issue of plateau when youre training with PointSetBC?

**Comment by guimilhoratti** [comment]
Hi, 
I believe my results are enough accurately to my problem (I hope so).
I did not test other changes. 

What about you?

**Reply by ryabs07** [reply]
PointSetBC is working for inverse problems for me. However, the training still plateaus for forward problems with semi-supervised training.

**Reply by guimilhoratti** [reply]
I see,

My case is an inverse problem, and PointsetBC() is working so far. I have not tryed a forward problem.

How many scattered points and data points are you using?

It would be good if we exchanged experiences. :)

Thank you
